src/preferences/dialog/dlgprefwaveform.cpp

```diff
@@ -196,7 +196,12 @@ void DlgPrefWaveform::slotSetWaveformType(int index) {
     if (index < 0) {
         return;
     }
+    // This regenerates twice the waveforms because of a bug found on Windows
+    // where the first one fails.
+    // The problem is that the window of the widget thinks that it is not exposed.
+    // (https://doc.qt.io/qt-5/qwindow.html#exposeEvent )
```

Comment:
```suggestion
    // This regenerates the waveforms twice because of a bug found on Windows
    // where the first one fails.
    // The problem is that the window of the widget thinks that it is not exposed.
    // (https://doc.qt.io/qt-5/qwindow.html#exposeEvent )
    // TODO: Remove this when it has been fixed upstream.
```

Reply:
Maybe add a link to a Qt bug report?

Reply:
I looked briefly but didn't find one.